tidb-performance-tuning-config.md

```diff
@@ -0,0 +1,527 @@
+
+- Use [`tidb_low_resolution_tso`](/system-variables.md#tidb_low_resolution_tso) for read operations that don't require strict consistency
+- Enable [`tidb_enable_batch_dml`](/system-variables.md#tidb_enable_batch_dml) to reduce TSO requests for batch operations
+- Consider batching small transactions into larger ones where possible
+
+#### Solution 1 `tidb_low_resolution_tso`
+
+The `tidb_low_resolution_tso` setting allows TiDB to use cached timestamps for read operations, significantly reducing TSO wait time at the cost of potentially stale reads. This optimization is particularly effective for:
+
+- Read-heavy workloads where slight staleness is acceptable
+- Scenarios where reducing query latency is more critical than absolute consistency
+- Applications that can tolerate reads that may be a few seconds behind the latest committed state
+
+To enable this optimization:
+
+```sql
+set global tidb_low_resolution_tso=on;
+```
+
+##### Justifications
+
+| configurations | Pro | Cons | 
+| ---------| ---- | ----|
+| [`tidb_low_resolution_tso`](system-variables.md#tidb_low_resolution_tso)| Reduces query latency by enabling stale reads with cached TSO, eliminating the need to request new timestamps | Trades consistency for performance - only suitable for scenarios that can tolerate stale reads. Not recommended when strict data consistency is required | 
+
+#### Solution 2 `tidb_tso_client_rpc_mode`
+
+The default value of `tidb_tso_client_rpc_mode` is `DEFAULT`. When the following conditions are met, you can consider switching `tidb_tso_client_rpc_mode` to `PARALLEL` or `PARALLEL-FAST` for potential performance improvements:
+
+- TSO waiting time constitutes a significant portion of the total execution time of SQL queries.
+- The TSO allocation in PD has not reached its bottleneck.
+- PD and TiDB nodes have sufficient CPU resources.
+- The network latency between TiDB and PD is significantly higher than the time PD takes to allocate TSO (that is, network latency accounts for the majority of TSO RPC duration).
+    - To get the duration of TSO RPC requests, check the PD TSO RPC Duration panel in the PD Client section of the TiDB dashboard.
+    - To get the duration of PD TSO allocation, check the PD server TSO handle duration panel in the TiDB section of the Grafana PD dashboard.
+- The additional network traffic resulting from more TSO RPC requests between TiDB and PD (twice for PARALLEL or four times for PARALLEL-FAST) is acceptable.
+
+```sql
+set global tidb_tso_client_rpc_mode=PARALLEL;
+```
```

Comment:
````suggestion
To switch the parallel mode, execute the following command:

```sql
-- Use the PARALLEL mode
SET GLOBAL tidb_tso_client_rpc_mode=PARALLEL;

-- Use the PARALLEL-FAST mode
SET GLOBAL tidb_tso_client_rpc_mode=PARALLEL-FAST;
```
````